lib/rubygems/commands/sources_command.rb

```diff
@@ -68,6 +76,63 @@ def add_source(source_uri) # :nodoc:
     end
   end
 
+  def append_source(source_uri) # :nodoc:
+    check_rubygems_https source_uri
+
+    source = Gem::Source.new source_uri
+
+    check_typo_squatting(source)
+
+    begin
+      was_present = Gem.sources.include?(source)
+
+      source.load_specs :released
+      Gem.sources.delete(source) if was_present
+      Gem.sources << source
```

Comment:
Should we code the source dance into `Gem::SourceList#append` and `Gem::SourceList#prepend`, and then DRY up both methods in to a single `add_source` method? We could rename current `add_source` to `deprecated_add_source` or something.

Also thinking about naming, should we instead use something like `ensure_first` or `ensure_last`? What people normally understands from `append` or `prepend` in other classes does not involve moving elements around, just appending or prepending.

Reply:
Good point. I'll get that.